app/controllers/api/v1/assignments_controller.rb

```diff
@@ -139,6 +138,23 @@ def copy_assignment
     end
   end
 
+  def has_badge
```

Comment:
Instead of using methods like has_badge, pair_programming_enabled, is_calibrated, staggered_and_no_topic to retrieve fields of assignment, you can create a show method in the controller that returns the assignment object in JSON format ( render json: course, status: :ok ), here, all the fields of the assignment model object are sent.